networkit/cpp/distance/MultiTargetBFS.cpp

```diff
@@ -61,6 +62,14 @@ void MultiTargetBFS::run() {
         toNextLevel();
     }
 
+    // Mark targets not reached as unreachable
+    while (!targetsSet.empty()) {
```

Comment:
Why not just iterating over targetsSet? This would remove the need for erasing the iterator in each loop run.